landa/patches/set_billing_and_shipping_defaults.py

```diff
@@ -0,0 +1,81 @@
+import json
+import os
+
+import frappe
+from frappe.modules.utils import sync_customizations_for_doctype
+from frappe.query_builder import DocType
+
+
+def execute():
+	set_billing_and_shipping_defaults()
+	cleanup_addresses()
+
+
+def customize_customer():
+	folder = frappe.get_app_path("landa", "landa_sales", "custom")
+	fname = "customer.json"
+
+	with open(os.path.join(folder, fname), "r") as f:
+		sync_customizations_for_doctype(json.loads(f.read()), folder)
+
+
+def set_billing_and_shipping_defaults():
+	customize_customer()
+
+	Address = DocType("Address")
+	DynamicLink = DocType("Dynamic Link")
+
+	addresses = (
+		frappe.qb.from_(Address)
+		.inner_join(DynamicLink)
+		.on(Address.name == DynamicLink.parent)
+		.select(
+			Address.name,
+			Address.is_shipping_address,
+			Address.is_primary_address,
+			DynamicLink.link_name,
+		)
+		.where((Address.is_shipping_address == 1) | (Address.is_primary_address == 1))
+		.where(DynamicLink.link_doctype == "Customer")
+		.run()
+	)
+
+	for address_name, is_shipping, is_primary, customer in addresses:
+		if is_primary:  # billing address
+			contact = frappe.get_all(
+				"Contact",
+				[["name", "like", address_name.replace(" ", "%")]],
+				pluck="name",
+			)
+
+			frappe.db.set_value(
+				"Customer",
+				customer,
+				{
+					"default_billing_contact": contact[0] if contact else None,
+					"default_billing_address": address_name,
+				},
+			)
+
+		if is_shipping:
+			contact = frappe.get_all(
+				"Contact",
+				[["name", "like", address_name.replace(" ", "%")]],
+				pluck="name",
+			)
+
+			frappe.db.set_value(
+				"Customer",
+				customer,
+				{
+					"default_shipping_contact": contact[0] if contact else None,
+					"default_shipping_address": address_name,
+				},
+			)
+
+
+def cleanup_addresses():
+	# the two checkboxes are no longer used and are hidden from now on.
+	frappe.db.sql(
+		"""update `tabAddress` set is_shipping_address=0, is_primary_address=0"""
+	)
```

Comment:
No harm in keeping this data as it is, right ? Just in case we need this data in the future.

Reply:
I don't see a point in keeping data which won't be consistent once a new address is added. Besides that, you can filter by the fields. This may lead to confusion along down the road.